zipkin-lens/src/components/App/LanguageSelector.tsx

```diff
@@ -63,15 +64,23 @@ const LanguageSelector = () => {
       if (!locale) {
         return;
       }
-      if (locale === currentLocale) {
+      if (locale === i18n.language) {
         return;
       }
+
       i18n.changeLanguage(locale);
-      setLocale(locale);
     },
-    [currentLocale, i18n],
+    [i18n],
   );
 
+  useEffect(() => {
+    if (LANGUAGES.find((lang) => lang.locale === i18n.language)) {
+      i18n.changeLanguage(i18n.language);
+    } else {
+      i18n.changeLanguage(FALLBACK_LOCALE); // fallback to default language if the selected language is not supported
+    }
+  }, []);
```

Comment:
```suggestion
  }, [i18n]);
```

Not sure but believe this is supposed to be here

Reply:
thanks, I'll give it a go locally

Reply:
I think the problem is that this will update the locale twice. So if I click on the button it will trigger i18next to change the locale. But because the locale is changed the useEffect is called and the language is again changed. The idea was to have this useEffect only be executed on page rendering.